src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -735,9 +735,9 @@ async Task ITokenCacheInternal.ClearAsync()
             {
                 TokenCacheNotificationArgs args = new TokenCacheNotificationArgs(this, ClientId, null, true, (this as ITokenCacheInternal).IsApplicationCache);
 
-                await (this as ITokenCacheInternal).OnBeforeAccessAsync(args).ConfigureAwait(false);
                 try
                 {
+                    await (this as ITokenCacheInternal).OnBeforeAccessAsync(args).ConfigureAwait(false);
```

Comment:
Moving these inside the try block because if the user adds a callback for OnBeforeAccess that throws exception, then catches them, MSAL will not fire `OnAfterAccessAsync`. This would break the extensions, as it would leave a permanent file lock on disk (and block VS, Powershell etc.)

CC @schaabs

Reply:
is this really doing what you expect? 

---
In reply to: [403098828](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1740#discussion_r403098828) [](ancestors = 403098828)

Reply:
It's a fair point, I think I can write a unit test that solidify this behaviour.